dali/image/tiff.cc

```diff
@@ -90,7 +81,55 @@ Image::ImageDims TiffImage::PeekDims(const uint8_t *encoded_buffer, size_t lengt
   }
 
   // TODO(mszolucha): fill channels count
-  return std::make_tuple(height, width, 0);
+  return {static_cast<int64_t>(height),
+          static_cast<int64_t>(width),
+          0};
+}
+
+}  // namespace legacy_impl
+
+TiffImage::TiffImage(const uint8_t *encoded_buffer, size_t length, dali::DALIImageType image_type) :
+        GenericImage(encoded_buffer, length, image_type) {
+#ifdef DALI_USE_LIBTIFF
+  libtiff_decoder_.reset(new LibtiffImpl(make_span(encoded_buffer, length)));
+#endif
+}
+
+Image::ImageDims TiffImage::PeekDims(const uint8_t *encoded_buffer, size_t length) const {
+  DALI_ENFORCE(encoded_buffer != nullptr);
+
+  kernels::TensorShape<3> shape;
+#ifdef DALI_USE_LIBTIFF
+  shape = libtiff_decoder_->Dims();
+#else
+  shape = legacy_impl::PeekDimsImpl(encoded_buffer, length);
+#endif
+
+  return std::make_tuple(
+    static_cast<size_t>(shape[0]),
```

Comment:
Use braced initializer instead of make_tuple?

Reply:
doesn't work. In any case, we are going to move from std::tuple to TensorShape soon

Reply:
@mzient see #1280